lib/chatter_web/live/chat_live/index.ex

```diff
@@ -42,24 +58,31 @@ defmodule ChatterWeb.ChatLive.Index do
 
   @impl true
   def handle_event("search", %{"search" => params}, socket) do
-    with {:ok, search} <- Search.create(params) do
+    with {:ok, search} <- Search.create(params),
+      search_messages <- user_shown_messages({search, socket}) do
+      broadcast_filtered_message(search_messages, socket.assigns.username)
```

Comment:
`socket.assigns.username` can be pattern matched in args

Reply:
changed to gather username with pattern matching